catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -150,9 +156,32 @@ public void init() {
   private CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
       throws UnsupportedQueryException, SourceUnavailableException, FederationException {
     QueryRequest request = cqlRequest.createQueryRequest(catalogFramework.getId(), filterBuilder);
-
     Stopwatch stopwatch = Stopwatch.createStarted();
-    QueryResponse response = catalogFramework.query(request);
+
+    List<QueryResponse> responses = Collections.synchronizedList(new ArrayList<>());
+    List<Result> results =
+        ResultIterable.resultIterable(
+                (queryRequest) -> {
+                  QueryResponse queryResponse = catalogFramework.query(queryRequest);
+                  responses.add(queryResponse);
+                  return queryResponse;
+                },
+                request,
+                cqlRequest.getCount())
+            .stream()
+            .collect(Collectors.toList());
+    QueryResponse response =
+        new QueryResponseImpl(
+            request,
+            results,
+            true,
+            responses.stream().map(QueryResponse::getHits).findFirst().orElse(-1l),
```

Comment:
All the responses should have the same value for hits. Take the first one.
Additionally, `CatalogFramework.query()` always returns at least one response. If those conditions hold, Line 178 collapses to:

`responses.get(0).getHits()`

Reply:
i prefer not to ever assume size and fetch by an index, this will always be safe no matter how implementations may change